### PR TITLE
fix: allow ts files to import js [sc-0]

### PR DESCRIPTION
**Pull request by umutuzgur**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

### Notes for the reviewer
* Allow more flexible setup for different extensions
* Allow TS files to import JS files
* Inline parsing errors to remove top try/catch

Closes https://github.com/checkly/checkly-cli/issues/423

### CODE REVIEW

File: package/src/services/check-parser/parser.ts [hunk 1]
```diff
@@ -46,89 +56,110 @@ export class Parser {
...
+        // @ts-ignore
+        walk.simple(ast, Parser.tsNodeVisitor(tsParser, localDependencies, npmDependencies))
+      } else {
+        throw new Error(`Unsupported file extension for ${filePath}`)
+      }
+    } catch (err) {
+      return {
+        localDependencies: Array.from(localDependencies),
+        npmDependencies: Array.from(npmDependencies),
+        error: err,
```
Comment (clample) [inline package/src/services/check-parser/parser.ts:161]: Very go-lang style 😄

File: package/src/services/check-parser/parser.ts [hunk 2]
```diff
@@ -46,89 +56,110 @@ export class Parser {
...
-    if (!filePath.endsWith(extension)) {
-      return filePath + extension
+  static readDependency (filePath: string, preferedExtenstion: SupportedExtensions) {
+    // Read the specific file if it has an extension
+    if (path.extname(filePath).length) {
+      const content = fs.readFileSync(filePath, { encoding: 'utf-8' })
+      return { filePath, content }
     } else {
-      return filePath
+      if (preferedExtenstion === SupportedExtensions.JS) {
```
Comment (clample) [inline package/src/services/check-parser/parser.ts:114]: Could it ever be the case that the main entrypoint check is in JS, but it imports TS files? Or is it only possible for TS to import JS?

File: package/src/services/check-parser/parser.ts (hunk 1 above)
Comment (umutuzgur) [reply package/src/services/check-parser/parser.ts:161]: Yeah. Error probe operation, like parsing, are really hard to achieve with the try/catch syntax

File: package/src/services/check-parser/parser.ts (hunk 2 above)
Comment (umutuzgur) [reply package/src/services/check-parser/parser.ts:114]: That shouldn't be allowed. We will experience probably different resolving issues in the future but js import ts in a pure matter shouldn't be possible

**Review by clample — APPROVED** [review]

**Pull request merged**